src/sentry/web/frontend/oauth_revoke.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from typing import Optional
+
+from django.db.models import Q
+from django.http import HttpRequest, HttpResponse
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import never_cache
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic.base import View
+
+from sentry.models import ApiApplication, ApiApplicationStatus, ApiToken
+from sentry.utils import json, metrics
+
+logger = logging.getLogger("sentry.api.oauth_revoke")
+
+
+class OAuthRevokeView(View):
+    """
+    OAuth 2.0 token revoke endpoint per RFC 7009
+    https://www.rfc-editor.org/rfc/rfc7009
+
+    Clients can provide either the access_token or refresh_token in the request.
+
+    When revoking the token, the associated access_token or refresh_token is also
+    revoked.
+    """
+
+    @csrf_exempt
+    @method_decorator(never_cache)
+    def dispatch(self, request, *args, **kwargs):
+        return super().dispatch(request, *args, **kwargs)
+
+    # Note: the reason parameter is for internal use only
+    def error(self, request: HttpRequest, name, reason=None, status=400):
+        client_id = request.POST.get("client_id")
+
+        logging.error(
```

Comment:
Can add any other additional fields to logging here, such as the user who made the request?

Are we also able to add logging in general when a request comes in and doesn't result in a error?

Reply:
It would be the client performing the revoke and not the user. What information would be most helpful here to log for your team?

I was thinking `client_id` and a hashed version of the `token`.

Reply:
I saw you already checked https://www.notion.so/sentry/Logging-Strategy-and-Guide-41d031bf2caf457d8eec79a8488102a5?pvs=4, but generally being able to answer who, what, when, and from where (IP or other location identifying attributes) is helpful. 

client_id looks like it's already in there. Hashed version of the `token` would be useful. 
Would it good to have a hashed version of the `refresh_token` if the `token` you are referring to is the `access_token`? 
what else can the client_id be tied to that could be useful (ie. application_id, org related info, owner of the token)?

Reply:
Added the additional fields. :)